uktrade/data-hub-frontend
# Add outline of pipelines app and add form

## Description of change

This PR adds the pipelines app and the UI for the add to pipeline form. This is the start of yellow team's mission work for this quarter to build functionality that allows users to add companies to three defined statuses - lead, in progress and win - in a single pipeline. This will work in a similar way to the existing company list feature.

This PR does not include the API calls as the endpoints aren't ready yet. These will be added in a follow up PR, so please view this as a work in progress. The url for this page is not linked anywhere on the site. 

The logic is that if a company is already in a pipeline it will show the user a message to that effect and links back to the company or to the dashboard. If the company isn't already in a pipeline it will show a short form where the user can select which status to assign the company - Lead, In Progress, or Win. 

## Test instructions

When you navigate to `/companies/0fb3379c-341c-4da4-b825-bf8d47b26baa/pipeline` (Lambda plc) you will see the message that the company is already on the pipeline. Any other company will show the form. 

## Screenshots

The form:
![image](https://user-images.githubusercontent.com/42253716/79993423-a96dad00-84ac-11ea-91fb-14507ef5dd34.png)

The 'already in pipeline' message:
![image](https://user-images.githubusercontent.com/42253716/79993313-88a55780-84ac-11ea-970d-56a5edd6f973.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
